panel/applets/status/PowerIndicator.vala

```diff
@@ -33,15 +35,43 @@ public class PowerIndicator : Gtk.Bin
 
     public PowerIndicator()
     {
-        batteries = new List<Up.Device>();
+        ebox = new Gtk.EventBox();
+        add(ebox);
 
+        batteries = new List<Up.Device>();
         widget = new Gtk.Box(Gtk.Orientation.HORIZONTAL, 2);
-        add(widget);
+        ebox.add(widget);
+
+        var menu = new GLib.Menu();
+        var power_settings = new GLib.MenuItem(_("Power settings"), "power.settings");
+        menu.append_item(power_settings);
+
+        popover = new Gtk.Popover.from_model(widget, menu);
```

Comment:
The popover is supposed to use the Gtk.EventBox, this is to catch the events and propagate them correctly